packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,138 @@ void main() {
     expect(value, equals('three'));
   });
 
+  testWidgets('DropdownButtonFormFieldState value updates when a different option is selected', (WidgetTester tester) async {
+    const String value = 'one';
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: value,
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets("DropdownButtonFormField's InputDecorator isEmpty behavior is correct", (WidgetTester tester) async {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: null, // setting value to null sets isEmpty to true
+            decoration: const InputDecoration(labelText: 'label', alignLabelWithHint: true),
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // isEmpty: true causes the label to be aligned with the input text if alignLabelWithHint is set to true
+    await tester.pumpAndSettle();
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 600.0));
```

Comment:
Knowing the size of the `InputDecorator` provides value for the test. Instead, there should be a comparison of the positioning of the `InputDecorator` with relation to the positioning of the label and checking if that is expected.

Same comment about the subsequent checks

Reply:
Something like 
```
expect(
  tester.getTopLeft(find.byType(InputDecorator)).dy, 
  tester.getTopLeft(find.text('label')).dy +/- {someValue},
); // but having a positioning (x and y) that makes sense and is expected. Explain in code comments why the offset or difference is expected
```

Reply:
I've added the check for that but it turns out that the value is the same as what's being tested by `expect(tester.getTopLeft(find.text('label')).dy, {y-value});`, so please remove as you see fit